lib/src/main/scala/spinal/lib/eda/fusesoc/FuseSocGeneratorBuilder.scala

```diff
@@ -0,0 +1,126 @@
+package spinal.lib.eda.fusesoc
+import spinal.core._
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.{YAMLFactory, YAMLGenerator}
+import com.fasterxml.jackson.module.scala._
+
+import scala.collection.immutable.Map.Map1
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+class GeneratorOutput() {
+  val parameters = mutable.Map[String, String]()
+  // List(file_name -> {file_type -> verilogSource})
+  val files = mutable.ArrayBuffer[Map1[String, Map[String, String]]]()
+}
+case class GeneratorParameters[P: ClassTag](
+    // for python only
```

Comment:
What does this comment for? Is this means there is some python part? How about refine it?

Reply:
It's hard to implement all things in spinal since fusesoc is a python stuff.

And refactoring this without rewrite serialize method will change *.core file structure, we don't want to leak inner implementation to core file user.(e.g. which field is implement by python or by spinal)